### PR TITLE
Working with custom routes file

### PR DESCRIPTION
**Pull request by FabioMR**
Sometimes I need to change the default routes path or define multiples files. So, I've adapted the code to this situation.

### CODE REVIEW

**Comment by le0pard** [comment]
Hello @FabioMR . To modify JS you should edit https://github.com/railsware/js-routes/blob/master/lib/routes.js.coffee instead of routes.js

**Comment by bogdan** [comment]
I didn't find any documentation on using `Rails.application.paths`. Can some suggest me a link?

**Comment by FabioMR** [comment]
@le0pard In fact, I didn't modified the routes.js. I just ran rspec and It was builded automatically based on .coffee file.

**Comment by FabioMR** [comment]
@bogdan You are already using Rails.application.paths in your code: https://github.com/railsware/js-routes/blob/master/spec/spec_helper.rb
I used the same idea, just changed the code to works for my context.

By the way, this is the guy: http://api.rubyonrails.org/classes/Rails/Paths/Path.html

**Comment by le0pard** [comment]
@FabioMR Ok, it is just new coffescript gem.

**Comment by FabioMR** [comment]
@le0pard Ok. But any problem? If you want, I can back routes.js changes.

**Comment by le0pard** [comment]
@bogdan will review PR, I just checked pull request on errors.

File: lib/js_routes/engine.rb [hunk 1]
```diff
@@ -4,10 +4,17 @@ class Engine < ::Rails::Engine
...
       if Rails.application.assets.respond_to?(:register_preprocessor)
-        routes = Rails.root.join('config','routes.rb')
-        Rails.application.assets.register_preprocessor 'application/javascript', :'js-routes_dependent_on_routes' do |ctx,data|
-          ctx.depend_on(routes) if ctx.logical_path == JS_ROUTES_ASSET
-          data
+        if Rails::VERSION::MAJOR == 3
+          route_paths = Rails.application.paths['config/routes'].to_a
+        else
+          route_paths = Rails.application.paths['config/routes.rb'].to_a
+        end
```
Comment (le0pard) [inline lib/js_routes/engine.rb]: `route_paths = Rails.application.paths["config/routes#{3 == Rails::VERSION::MAJOR ? '' : '.rb'}"].to_a`

Comment (FabioMR) [inline lib/js_routes/engine.rb]: Ok, less messy! I'll change it!

**Comment by bogdan** [comment]
I don't see any doc on how `paths` should be used. That's why it looks strange to me that you are using `paths` in an undocumented way (maybe along with js-routes test suite - but it is ok there because test suite runtime environment is a hack). 

Maybe this patch makes it work for your case and break it for other person case. 
The `Paths::Path` didn't give an idea how I can use `Rails.application.paths`.  It would be cool to see Rails doc or how-to of how the `paths` are used.

**Comment by FabioMR** [comment]
@bogdan I can't see how this patch can break in some cases, because it is how Rails use the routes. But I agree that the doc is very poor. I think I get your point.

But how can we go with this? I could change the route path through a config, what do your think? The config could have a default value and we can change it if necessary.

**Comment by bogdan** [comment]
Here is what I've found: https://github.com/rails/rails/commit/5e7d6bba79393de0279917f93b82f3b7b176f4b5
https://github.com/rails/rails/compare/fa736e69a197...2d9dbf416b14

It looks like this feature works only in 3.x.x series and you are forced to use concerns in 4.x.x

**Comment by FabioMR** [comment]
I tested my project with the original config/routes.rb and just imported other files. It worked fine too. So I think this pull request can be closed, it is not necessary anymore!

I hope this issue could help others with the same problem as I.

Thank you!

**Solution:**
`#config/routes.rb`
`require_relative 'routes/application.rb'`
`require_relative 'routes/api.rb'`

**Pull request closed without merging**